client/Cargo.toml

```diff
@@ -11,7 +11,7 @@ license = "Apache-2.0 OR Zlib"
 [dependencies]
 common = { path = "../common" }
 server = { path = "../server" }
-tracing = "0.1.10"
+tracing = "0.1.36"
```

Comment:
There's rarely any need to update patch versions manually; newer patch versions are always considered compatible and preferred.

Reply:
Thanks for pointing that out. I had known that before, but I must have missed that for `tracing`. Should I revert this change, or leave it as-is, since it's already part of the PR?

Reply:
It's harmless, so I don't really care, but it should also be very easy to drop via interactive rebase. I'm assuming this PR is subject to change regardless as you have it marked as a draft.

Reply:
Makes sense to me. I believe the PR is mostly complete, as the only change that's left is to upgrade the version of nalgebra. Everything should be fair game for review. I'll go ahead and do the rebase.